Tella/Scenes/Resources/ViewModel/AvailableResourcesVM.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  AvailableResourcesVM.swift
+//  Tella
+//
+//  Created by gus valbuena on 2/5/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Combine
+import Foundation
+import Combine
+
+class AvailableResourcesVM: ObservableObject {
+    @Published var appModel: MainAppModel
+    @Published var availableResources: [ResourceCardViewModel] = []
+    @Published var downloadedResourcesVM: DownloadedResourcesVM?
+    @Published var isLoadingList: Bool = false
+    @Published var isLoadingDownload: Bool = false
+    @Published var servers: [TellaServer] = []
+    private let resourceService: ResourceService
+    private var cancellables: Set<AnyCancellable> = []
+
+    init(
+        mainAppModel: MainAppModel, downloadedVM: DownloadedResourcesVM? = nil,
+        resourceService: ResourceService = ResourceService()
+    ) {
+        self.appModel = mainAppModel
+        self.servers = mainAppModel.vaultManager.tellaData?.tellaServers.value ?? []
+        self.downloadedResourcesVM = downloadedVM
+        self.resourceService = resourceService
+        self.getAvailableForDownloadResources()
+        self.listenDownloadDeletion()
+    }
+
+    func getAvailableForDownloadResources() {
+        self.isLoadingList = true
+        self.availableResources = []
+
+        resourceService.getAvailableResources(appModel: appModel, servers: servers)
+            .receive(on: DispatchQueue.main)
+            .sink(
+                receiveCompletion: { completion in
+                    switch completion {
+                    case .finished:
+                        self.isLoadingList = false
+                        break
+                    case .failure(let error):
+                        self.isLoadingList = false
+                        Toast.displayToast(message: LocalizableResources.resourcesAvailableErrorMsg.localized)
+                        print("Error: \(error)")
+                    }
+
+                },
+                receiveValue: { resources in
+                    self.isLoadingList = false
+                    resources.forEach { resource in
+                        let isDownloaded = self.downloadedResourcesVM?.downloadedResources.contains(where: { $0.externalId == resource.id })
```

Comment:
yea you're right 
maybe we can try this code  if it works ?
```
self.availableResources = resources.filter { (resource) in
self.downloadedResourcesVM?.downloadedResources.contains{$0.externalId == resource.id} ?? false
}

```

Reply:
I changed it to this code to avoid the loop on resources:

receiveValue: { resources in
                    self.isLoadingList = false
                    let downloadedIds = Set(self.downloadedResources.map { $0.externalId })
                    let newResources = resources.filter { !downloadedIds.contains($0.id) }
                    self.availableResources.append(contentsOf: newResources)

                }